diff --git fastfetch-clean/CMakeLists.txt fastfetch-workdir/CMakeLists.txt
index 2c6c8bd..86b97db 100644
--- fastfetch-clean/CMakeLists.txt
+++ fastfetch-workdir/CMakeLists.txt
@@ -45,6 +45,8 @@ elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Haiku")
     set(Haiku TRUE CACHE BOOL "..." FORCE)
 elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "GNU")
     set(GNU TRUE CACHE BOOL "..." FORCE)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "bentobox")
+    set(BENTOBOX TRUE CACHE BOOL "..." FORCE)
 elseif(NOT APPLE AND NOT WIN32)
     message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
 endif()
@@ -1324,6 +1326,87 @@ elseif(GNU)
         src/detection/wmtheme/wmtheme_linux.c
         src/detection/camera/camera_nosupport.c
    )
+elseif(BENTOBOX)
+    list(APPEND LIBFASTFETCH_SRC
+        src/common/impl/dbus.c
+        src/common/impl/io_unix.c
+        src/common/impl/netif_gnu.c
+        src/common/impl/networking_linux.c
+        src/common/impl/processing_linux.c
+        src/common/impl/FFPlatform_unix.c
+        src/common/impl/binary_linux.c
+        src/common/impl/kmod_linux.c
+        src/detection/battery/battery_linux.c
+        src/detection/bios/bios_nosupport.c
+        src/detection/board/board_nosupport.c
+        src/detection/bootmgr/bootmgr_nosupport.c
+        src/detection/brightness/brightness_nosupport.c
+        src/detection/btrfs/btrfs_nosupport.c
+        src/detection/chassis/chassis_nosupport.c
+        src/detection/cpu/cpu_nosupport.c
+        src/detection/cpucache/cpucache_nosupport.c
+        src/detection/cpuusage/cpuusage_nosupport.c
+        src/detection/cursor/cursor_nosupport.c
+        src/detection/bluetooth/bluetooth_nosupport.c
+        src/detection/bluetoothradio/bluetoothradio_nosupport.c
+        src/detection/disk/disk_nosupport.c
+        src/detection/dns/dns_linux.c
+        src/detection/physicaldisk/physicaldisk_nosupport.c
+        src/detection/physicalmemory/physicalmemory_nosupport.c
+        src/detection/diskio/diskio_nosupport.c
+        src/detection/displayserver/linux/displayserver_linux.c
+        src/detection/displayserver/linux/common.c
+        src/detection/displayserver/linux/drm.c
+        src/detection/displayserver/linux/wayland/wayland.c
+        src/detection/displayserver/linux/wayland/global-output.c
+        src/detection/displayserver/linux/wayland/zwlr-output.c
+        src/detection/displayserver/linux/wayland/kde-output.c
+        src/detection/displayserver/linux/wayland/wlr-output-management-unstable-v1-protocol.c
+        src/detection/displayserver/linux/wayland/kde-output-device-v2-protocol.c
+        src/detection/displayserver/linux/wayland/kde-output-order-v1-protocol.c
+        src/detection/displayserver/linux/wayland/xdg-output-unstable-v1-protocol.c
+        src/detection/displayserver/linux/wmde.c
+        src/detection/displayserver/linux/xcb.c
+        src/detection/displayserver/linux/xlib.c
+        src/detection/font/font_linux.c
+        src/detection/gpu/gpu_nosupport.c
+        src/detection/gtk_qt/gtk.c
+        src/detection/host/host_nosupport.c
+        src/detection/icons/icons_linux.c
+        src/detection/initsystem/initsystem_linux.c
+        src/detection/keyboard/keyboard_nosupport.c
+        src/detection/libc/libc_linux.c
+        src/detection/lm/lm_nosupport.c
+        src/detection/loadavg/loadavg_nosupport.c
+        src/detection/locale/locale_linux.c
+        src/detection/localip/localip_nosupport.c
+        src/detection/gamepad/gamepad_nosupport.c
+        src/detection/media/media_nosupport.c
+        src/detection/memory/memory_linux.c
+        src/detection/mouse/mouse_nosupport.c
+        src/detection/netio/netio_nosupport.c
+        src/detection/opengl/opengl_linux.c
+        src/detection/os/os_linux.c
+        src/detection/packages/packages_nosupport.c
+        src/detection/poweradapter/poweradapter_linux.c
+        src/detection/processes/processes_linux.c
+        src/detection/gtk_qt/qt.c
+        src/detection/sound/sound_nosupport.c
+        src/detection/swap/swap_linux.c
+        src/detection/terminalfont/terminalfont_linux.c
+        src/detection/terminalshell/terminalshell_linux.c
+        src/detection/terminalsize/terminalsize_linux.c
+        src/detection/theme/theme_linux.c
+        src/detection/tpm/tpm_nosupport.c
+        src/detection/uptime/uptime_linux.c
+        src/detection/users/users_linux.c
+        src/detection/wallpaper/wallpaper_nosupport.c
+        src/detection/wifi/wifi_nosupport.c
+        src/detection/wm/wm_linux.c
+        src/detection/de/de_linux.c
+        src/detection/wmtheme/wmtheme_linux.c
+        src/detection/camera/camera_nosupport.c
+    )
 endif()
 
 # Proprietary GPU driver APIs
diff --git fastfetch-clean/src/common/impl/FFPlatform_unix.c fastfetch-workdir/src/common/impl/FFPlatform_unix.c
index 8320a17..8b5d8fe 100644
--- fastfetch-clean/src/common/impl/FFPlatform_unix.c
+++ fastfetch-workdir/src/common/impl/FFPlatform_unix.c
@@ -141,6 +141,8 @@ static void getExePath(FFPlatform* platform) {
             break;
         }
     }
+#elif defined(__bentobox__)
+    size_t exePathLen = 0;
 #endif
     if (exePathLen > 0) {
         ffStrbufEnsureFree(&platform->exePath, PATH_MAX);
diff --git fastfetch-clean/src/detection/displayserver/linux/displayserver_linux.c fastfetch-workdir/src/detection/displayserver/linux/displayserver_linux.c
index 174aeb3..54e376a 100644
--- fastfetch-clean/src/detection/displayserver/linux/displayserver_linux.c
+++ fastfetch-workdir/src/detection/displayserver/linux/displayserver_linux.c
@@ -41,6 +41,7 @@ static void getWMProtocolNameFromEnv(FFDisplayServerResult* result) {
 }
 
 void ffConnectDisplayServerImpl(FFDisplayServerResult* ds) {
+    #ifndef __bentobox__
     if (instance.config.general.dsForceDrm == FF_DS_FORCE_DRM_TYPE_FALSE) {
         // We try wayland as our preferred display server, as it supports the most features.
         // This method can't detect the name of our WM / DE
@@ -57,6 +58,7 @@ void ffConnectDisplayServerImpl(FFDisplayServerResult* ds) {
             ffdsConnectXrandr(ds);
         }
     }
+    #endif
 
     // This display detection method is display server independent.
     // Use it if all connections failed
diff --git fastfetch-workdir/src/detection/localip/localip_nosupport.c fastfetch-workdir/src/detection/localip/localip_nosupport.c
new file mode 100644
index 0000000..4948730
--- /dev/null
+++ fastfetch-workdir/src/detection/localip/localip_nosupport.c
@@ -0,0 +1,8 @@
+#include "localip.h"
+
+const char* ffDetectLocalIps(const FFLocalIpOptions* options, FFlist* results)
+{
+    (void)options;
+    (void)results;
+    return "Local IP detection is not supported on this platform";
+}
\ No newline at end of file
diff --git fastfetch-clean/src/detection/sound/sound_nosupport.c fastfetch-workdir/src/detection/sound/sound_nosupport.c
index b672360..cc0ade5 100644
--- fastfetch-clean/src/detection/sound/sound_nosupport.c
+++ fastfetch-workdir/src/detection/sound/sound_nosupport.c
@@ -1,5 +1,5 @@
 #include "sound.h"
 
-const char* ffDetectSound(FF_A_UNUSED FFlist* devices /* List of FFSoundDevice */) {
+const char* ffDetectSound(FF_A_UNUSED FFSoundOptions* options, FF_A_UNUSED FFlist* devices /* List of FFSoundDevice */) {
     return "Not supported on this platform";
 }
