diff --git glib2-clean/glib/glib-unix.c glib2-workdir/glib/glib-unix.c
index 24a20df..9dbf2f9 100644
--- glib2-clean/glib/glib-unix.c
+++ glib2-workdir/glib/glib-unix.c
@@ -1006,6 +1006,10 @@ g_unix_fd_query_path (int      fd,
   g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOSYS,
                "g_unix_fd_query_path() not supported on HURD");
   return NULL;
+#elif defined (__bentobox__)
+  g_set_error (error, G_FILE_ERROR, G_FILE_ERROR_NOSYS,
+               "g_unix_fd_query_path() not supported on bentobox");
+  return NULL;
 #else
   #error "g_unix_fd_query_path() not supported on this platform"
 #endif
