diff --git yajl-clean/reformatter/CMakeLists.txt yajl-workdir/reformatter/CMakeLists.txt
index 52a9bee..8fe05ec 100644
--- yajl-clean/reformatter/CMakeLists.txt
+++ yajl-workdir/reformatter/CMakeLists.txt
@@ -35,9 +35,9 @@ IF (NOT WIN32)
 ENDIF (NOT WIN32)
 
 # copy the binary into the output directory
-GET_TARGET_PROPERTY(binPath json_reformat LOCATION)
+set(JSON_REFORMAT_BIN $<TARGET_FILE:json_reformat>)
 
 ADD_CUSTOM_COMMAND(TARGET json_reformat POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${binPath} ${binDir})
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:json_reformat> ${binDir})
 
 INSTALL(TARGETS json_reformat RUNTIME DESTINATION bin)
diff --git yajl-clean/verify/CMakeLists.txt yajl-workdir/verify/CMakeLists.txt
index 967fca1..48a94be 100644
--- yajl-clean/verify/CMakeLists.txt
+++ yajl-workdir/verify/CMakeLists.txt
@@ -29,9 +29,9 @@ ADD_EXECUTABLE(json_verify ${SRCS})
 TARGET_LINK_LIBRARIES(json_verify yajl_s)
 
 # copy in the binary
-GET_TARGET_PROPERTY(binPath json_verify LOCATION)
+set(JSON_VERIFY_BIN $<TARGET_FILE:json_verify>)
 
 ADD_CUSTOM_COMMAND(TARGET json_verify POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${binPath} ${binDir})
+    COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:json_verify> ${binDir})
 
 INSTALL(TARGETS json_verify RUNTIME DESTINATION bin)
