--- jinx_original	2026-04-25 17:16:28.724947119 +0200
+++ jinx	2026-04-25 17:17:32.367034760 +0200
@@ -83,10 +83,6 @@
         die "$0: Current directory already initialised, run '$0 deinit' first"
     fi
 
-    if [ -f "Jinxfile" ]; then
-        die "$0: In-tree builds are not supported"
-    fi
-
     cat <<EOF >.jinx-parameters
 #! /bin/sh
 
@@ -639,7 +635,7 @@
         mkdir -p "${imgroot}/sysroot"
 
         container_pkgs_native_mount="mount -o ro --bind '${container_pkgs}' '${imgroot}/usr/local'"
-        sysroot_native_mount="mount -o ro --bind '${sysroot_dir}' '${imgroot}/sysroot'"
+        sysroot_native_mount="mount --bind '${sysroot_dir}' '${imgroot}/sysroot'"
 
         run_in_cont_lang=en_US.UTF-8
     else
@@ -1229,7 +1225,7 @@
     cross_compile=yes
     prepare_container
 
-    container_sources_ro="-o ro"
+    container_sources_ro=""
     if ! [ -d "${build_dir}"/host-builds/$1 ]; then
         run_in_container internal-configure-host $1
     fi
@@ -1286,7 +1282,7 @@
 
     prepare_container
 
-    container_sources_ro="-o ro"
+    container_sources_ro=""
     if ! [ -d "${build_dir}"/builds/$1 ]; then
         run_in_container internal-configure $1
     fi
