diff --git a/Makefile b/Makefile
index e92bcaf..4a6fcba 100644
--- a/Makefile
+++ b/Makefile
@@ -68,7 +68,7 @@ DFILES	= Makefile Makefile.tc $(MANUAL) $(OBJS:.o=.c) chkfont.c getopt.c \
 all: $(BINS)
 
 figlet: $(OBJS)
-	$(LD) $(LDFLAGS) -o $@ $(OBJS)
+	$(LD) $(OBJS) $(LDFLAGS) -o $@
 
 chkfont: chkfont.o
 	$(LD) $(LDFLAGS) -o $@ chkfont.o
diff --git a/utf8.h b/utf8.h
index a3e09bd..00c4f86 100644
--- a/utf8.h
+++ b/utf8.h
@@ -22,6 +22,16 @@
 
 #include <sys/types.h>
 
+#ifndef __BEGIN_DECLS
+#ifdef __cplusplus
+#define __BEGIN_DECLS extern "C" {
+#define __END_DECLS }
+#else
+#define __BEGIN_DECLS
+#define __END_DECLS
+#endif
+#endif
+
 #include <wchar.h>
 
 #define UTF8_IGNORE_ERROR		0x01
