--- config.sub	2020-12-31 15:34:29.000000000 +0100
+++ confignew.sub	2025-12-09 22:01:46.259645580 +0100
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -bentobox* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
