diff --git a/Makefile b/Makefile
index 15db421..51f44cd 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@ install: st
 	mkdir -p $(DESTDIR)$(MANPREFIX)/man1
 	sed "s/VERSION/$(VERSION)/g" < st.1 > $(DESTDIR)$(MANPREFIX)/man1/st.1
 	chmod 644 $(DESTDIR)$(MANPREFIX)/man1/st.1
-	tic -sx st.info
+	tic -sx -o $(DESTDIR)$(PREFIX)/share/terminfo st.info
 	@echo Please see the README file regarding the terminfo entry of st.
 
 uninstall:
diff --git a/config.def.h b/config.def.h
index 2cd740a..fab578d 100644
--- a/config.def.h
+++ b/config.def.h
@@ -5,7 +5,7 @@
  *
  * font: see http://freedesktop.org/software/fontconfig/fontconfig-user.html
  */
-static char *font = "Liberation Mono:pixelsize=12:antialias=true:autohint=true";
+static char *font = "IBM Plex Mono:style=Medium:pixelsize=13:antialias=true:autohint=true";
 static int borderpx = 2;
 
 /*
diff --git a/config.mk b/config.mk
index 2fc854e..13c295d 100644
--- a/config.mk
+++ b/config.mk
@@ -4,11 +4,11 @@ VERSION = 0.9.3
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = $(BASE)/usr
 MANPREFIX = $(PREFIX)/share/man
 
-X11INC = /usr/X11R6/include
-X11LIB = /usr/X11R6/lib
+X11INC = $(BASE)/usr/X11R6/include
+X11LIB = $(BASE)/usr/X11R6/lib
 
 PKG_CONFIG = pkg-config
 
@@ -21,7 +21,7 @@ LIBS = -L$(X11LIB) -lm -lrt -lX11 -lutil -lXft \
        `$(PKG_CONFIG) --libs freetype2`
 
 # flags
-STCPPFLAGS = -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600
+STCPPFLAGS = -DVERSION=\"$(VERSION)\" -D_XOPEN_SOURCE=600 -D_DEFAULT_SOURCE
 STCFLAGS = $(INCS) $(STCPPFLAGS) $(CPPFLAGS) $(CFLAGS)
 STLDFLAGS = $(LIBS) $(LDFLAGS)
 
@@ -33,4 +33,4 @@ STLDFLAGS = $(LIBS) $(LDFLAGS)
 #MANPREFIX = ${PREFIX}/man
 
 # compiler and linker
-# CC = c99
+CC = $(ARCH)-pc-bentobox-gcc
diff --git a/st.c b/st.c
index 6f40e35..956c060 100644
--- a/st.c
+++ b/st.c
@@ -20,7 +20,7 @@
 #include "st.h"
 #include "win.h"
 
-#if   defined(__linux)
+#if   defined(__linux) || defined(__bentobox__)
  #include <pty.h>
 #elif defined(__OpenBSD__) || defined(__NetBSD__) || defined(__APPLE__)
  #include <util.h>
