diff --git a/configure.ac b/configure.ac
index a800105..2ad10da 100644
--- a/configure.ac
+++ b/configure.ac
@@ -83,6 +83,9 @@ case $host_os in
   gnu*)
     OS_MOUSE_NAME=hurd
     ;;
+  bentobox*)
+    OS_MOUSE_NAME=bentobox
+    ;;
 esac
 AC_SUBST([OS_MOUSE_NAME])
 
diff --git a/src/bentobox_mouse.c b/src/bentobox_mouse.c
new file mode 100644
index 0000000..20b5eba
--- /dev/null
+++ b/src/bentobox_mouse.c
@@ -0,0 +1,132 @@
+#ifdef HAVE_XORG_CONFIG_H
+#include <xorg-config.h>
+#endif
+
+#include <xorg-server.h>
+#include <X11/X.h>
+#include <X11/Xproto.h>
+#include "inputstr.h"
+#include "scrnintstr.h"
+#include "mipointer.h"
+
+#include "xf86.h"
+#include "xf86Xinput.h"
+#include "mouse.h"
+#include "xf86_OSlib.h"
+#include "xisb.h"
+
+#include <stdio.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <linux/input.h>
+
+#define DEVICE_PATH "/dev/event1"
+
+static void bentoboxReadInput(InputInfoPtr pInfo) {
+    MouseDevPtr mouse = pInfo->private;
+    struct input_event ev;
+    int fd = pInfo->fd;
+    
+    int dx = 0, dy = 0;
+    bool has_motion = false;
+
+    while (read(fd, &ev, sizeof(ev)) == sizeof(ev)) {
+        if (ev.type == EV_REL) {
+            if (ev.code == REL_X) {
+                dx += ev.value;
+                has_motion = true;
+            } else if (ev.code == REL_Y) {
+                dy += ev.value;
+                has_motion = true;
+            }
+        } else if (ev.type == EV_KEY) {
+            if (ev.code >= BTN_LEFT && ev.code < BTN_LEFT + MAX_BUTTONS) {
+                int button = ev.code - BTN_LEFT + 1;
+                unsigned int bit = 1 << button;
+
+                if (ev.value)
+                    mouse->lastButtons |= bit;
+                else
+                    mouse->lastButtons &= ~bit;
+
+                mouse->PostEvent(pInfo, mouse->lastButtons, 0, 0, 0, 0);
+            }
+        } else if (ev.type == EV_SYN && ev.code == SYN_REPORT) {
+            if (has_motion) {
+                mouse->PostEvent(pInfo, mouse->lastButtons, dx, -dy, 0, 0);
+                dx = dy = 0;
+                has_motion = false;
+            }
+        }
+    }
+}
+
+static Bool bentoboxPreInit(InputInfoPtr pInfo, const char *proto, int flags) {
+    MouseDevPtr mouse;
+    mouse = pInfo->private;
+    mouse->protocol = proto;
+    xf86ProcessCommonOptions(pInfo, pInfo->options);
+
+    pInfo->fd = xf86OpenSerial(pInfo->options);
+    if (pInfo->fd < 0) {
+        xf86Msg(X_ERROR, "Failed to open %s: %s\n", DEVICE_PATH, strerror(errno));
+        return FALSE;
+    }
+
+    mouse->CommonOptions(pInfo);
+    pInfo->read_input = bentoboxReadInput;
+    
+    return TRUE;
+}
+
+static const char *getdevice(InputInfoPtr pInfo, const char *proto, int flag) {
+    struct stat st;
+
+    if (stat(DEVICE_PATH, &st) < 0)
+        return NULL;
+
+    pInfo->options = xf86AddNewOption(pInfo->options, "Device", DEVICE_PATH);
+
+	return DEVICE_PATH;
+}
+
+static int interfacesupport() {
+	return MSE_PS2;
+}
+
+static const char *names[] = {
+	"BentoboxMouseDev", NULL
+};
+
+static const char **protonames() {
+	return names;
+}
+
+static const char *defaultproto() {
+	return names[0];
+}
+
+static Bool isdefault(const char *proto) {
+	return strcmp(proto, defaultproto()) == 0;
+}
+
+OSMouseInfoPtr OSMouseInit(int flag) {
+	OSMouseInfoPtr p;
+
+	p = calloc(sizeof(OSMouseInfoRec), 1);
+
+	if (!p)
+		return NULL;
+
+	p->SupportedInterfaces = interfacesupport;
+	p->BuiltinNames = protonames;
+	p->FindDevice = getdevice;
+	p->DefaultProtocol = defaultproto;
+	p->CheckProtocol = isdefault;
+	p->PreInit = bentoboxPreInit;
+	return p;
+}
+
