diff --git nes-clean/CMakeLists.txt nes-workdir/CMakeLists.txt
index ac373de..ecfaf4b 100644
--- nes-clean/CMakeLists.txt
+++ nes-workdir/CMakeLists.txt
@@ -8,8 +8,6 @@ if(MSVC)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
 endif()
 
-set(SDL_DIR ${CMAKE_SOURCE_DIR}/external/SDL)
-set(SDL_ttf_DIR ${CMAKE_SOURCE_DIR}/external/SDL_ttf)
 option(TTF_WITH_HARFBUZZ "use harfbuzz to improve text shaping" ON)
 
 IF(ANDROID)
@@ -22,8 +20,9 @@ ELSE()
   option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
 ENDIF()
 
-add_subdirectory(${SDL_DIR})
-add_subdirectory(${SDL_ttf_DIR})
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(SDL2 REQUIRED sdl2)
+pkg_check_modules(SDL2_ttf REQUIRED SDL2_ttf)
 
 set (SRC
         src/apu.c
@@ -56,7 +55,7 @@ set (SRC
         src/biquad.h
         src/font.h
 )
-include_directories(src src/mappers vendor/include ${SDL_DIR}/include ${SDL_ttf_DIR})
+include_directories(src src/mappers vendor/include ${SDL2_INCLUDE_DIRS} ${SDL2_ttf_INCLUDE_DIRS})
 
 
 IF(ANDROID)
@@ -79,7 +78,7 @@ ELSE()
   ELSE()
     add_executable(nes ${SRC})
   ENDIF ()
-  target_link_libraries(nes SDL2::SDL2main SDL2::SDL2-static SDL2_ttf::SDL2_ttf-static)
+  target_link_libraries(nes ${SDL2_LIBRARIES} ${SDL2_ttf_LIBRARIES} m)
 ENDIF()
 
 IF (NOT WIN32)
diff --git nes-clean/src/apu.c nes-workdir/src/apu.c
index d192d17..e56b510 100644
--- nes-clean/src/apu.c
+++ nes-workdir/src/apu.c
@@ -171,10 +171,6 @@ void init_audio_device(const APU* apu) {
     want.silence = 0;
 
     apu->emulator->g_ctx.audio_device = SDL_OpenAudioDevice(NULL, 0, &want, NULL, 0);
-    if (apu->emulator->g_ctx.audio_device == 0) {
-        LOG(ERROR , SDL_GetError());
-        quit(EXIT_FAILURE);
-    }
 }
 
 void exit_APU() {
diff --git nes-clean/src/controller.c nes-workdir/src/controller.c
index c70f042..6129dbe 100644
--- nes-clean/src/controller.c
+++ nes-workdir/src/controller.c
@@ -48,10 +48,10 @@ void keyboard_mapper(struct JoyPad* joyPad, SDL_Event* event){
         case SDLK_RSHIFT:
             key = SELECT;
             break;
-        case SDLK_j:
+        case SDLK_a:
             key = BUTTON_A;
             break;
-        case SDLK_k:
+        case SDLK_s:
             key = BUTTON_B;
             break;
         case SDLK_l:
diff --git nes-clean/src/gfx.c nes-workdir/src/gfx.c
index a8b91fd..3375728 100644
--- nes-clean/src/gfx.c
+++ nes-workdir/src/gfx.c
@@ -10,7 +10,7 @@
 
 void get_graphics_context(GraphicsContext* ctx){
 
-    SDL_Init(SDL_INIT_EVERYTHING);
+    SDL_Init(SDL_INIT_EVERYTHING & ~SDL_INIT_AUDIO);
     TTF_Init();
 #ifdef __ANDROID__
     ctx->font = TTF_OpenFont("asap.ttf", (int)(ctx->screen_height * 0.05));
@@ -41,7 +41,6 @@ void get_graphics_context(GraphicsContext* ctx){
         ctx->width * (int)ctx->scale,
         ctx->height * (int)ctx->scale,
         SDL_WINDOW_SHOWN
-        | SDL_WINDOW_OPENGL
         | SDL_WINDOW_RESIZABLE
         | SDL_WINDOW_ALLOW_HIGHDPI
     );
@@ -53,7 +52,7 @@ void get_graphics_context(GraphicsContext* ctx){
     }
     SDL_SetWindowMinimumSize(ctx->window, ctx->width, ctx->height);
 
-    ctx->renderer = SDL_CreateRenderer(ctx->window, -1, SDL_RENDERER_ACCELERATED);
+    ctx->renderer = SDL_CreateRenderer(ctx->window, -1, 0);
     if(ctx->renderer == NULL){
         LOG(ERROR, SDL_GetError());
         quit(EXIT_FAILURE);
diff --git nes-clean/src/nsf.c nes-workdir/src/nsf.c
index 4313578..250d2f1 100644
--- nes-clean/src/nsf.c
+++ nes-workdir/src/nsf.c
@@ -16,12 +16,12 @@
 #define BAR_COUNT 128
 #define MAX_SILENCE 150 // frames
 
-static uint8_t read_PRG(const Mapper*, uint16_t);
+static uint8_t read_PRG(Mapper*, uint16_t);
 static void write_PRG(Mapper*, uint16_t, uint8_t);
 static uint8_t read_CHR(Mapper*, uint16_t);
 static void write_CHR(Mapper*, uint16_t, uint8_t);
 static uint8_t read_ROM(Mapper*, uint16_t);
-static void write_ROM(const Mapper*, uint16_t, uint8_t);
+static void write_ROM(Mapper*, uint16_t, uint8_t);
 
 static void read_text_stream(char** list, const char* buf, size_t list_len, size_t buf_len, size_t max_str_len);
 static void load_info_chunk(uint32_t len, Mapper* mapper, SDL_RWops* file);
@@ -455,7 +455,7 @@ static uint8_t read_ROM(Mapper* mapper, uint16_t addr) {
     return mapper->read_PRG(mapper, addr);
 }
 
-static void write_ROM(const Mapper* mapper, uint16_t addr, uint8_t val) {
+static void write_ROM(Mapper* mapper, uint16_t addr, uint8_t val) {
     if(addr < 0x6000) {
         if(addr > 0x5ff7) {
             mapper->NSF->bank_ptrs[addr - 0x5ff8] = mapper->PRG_ROM + val * 0x1000;
@@ -465,7 +465,7 @@ static void write_ROM(const Mapper* mapper, uint16_t addr, uint8_t val) {
     }
 }
 
-static uint8_t read_PRG(const Mapper* mapper, uint16_t addr) {
+static uint8_t read_PRG(Mapper* mapper, uint16_t addr) {
     if(!mapper->NSF->bank_switch) {
         return mapper->PRG_ROM[addr - 0x8000];
     }
